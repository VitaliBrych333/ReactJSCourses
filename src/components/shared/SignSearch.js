import React from 'react';
import { MDBIcon } from "mdbreact";
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const StyledSpan = styled.span`
    display: flex;
    justify-content: flex-end;
    padding-right: 25px;

    i {
        margin-top: 25px;
        transform: rotateZ(90deg);
        color: red;

        &:hover {
            cursor: pointer;
            color: gray;
        }
    }
`;

const SignSearch = () => {
    return (
        <StyledSpan>
            <Link to={{pathname: '/search/Search20Query'}}><MDBIcon icon="search" size="1x" className="pr-3" /></Link>
        </StyledSpan>
    );
};

export default SignSearch;
