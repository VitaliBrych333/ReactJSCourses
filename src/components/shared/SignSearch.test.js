import React from 'react';
import ReactDOM, { unmountComponentAtNode } from 'react-dom';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SignSearch from './SignSearch';

Enzyme.configure({ adapter: new Adapter() });

describe('<SignSearch/>', () => {
  let container = null;
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('should render the component', () => {
    ReactDOM.render(
      <Router>
        <SignSearch />
      </Router>,
      container
    );
    ReactDOM.unmountComponentAtNode(container);
  });

  it('should equals to snapshot of SignSearch', () => {
    const renderedValue = renderer
      .create(
        <Router>
          <SignSearch />
        </Router>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
