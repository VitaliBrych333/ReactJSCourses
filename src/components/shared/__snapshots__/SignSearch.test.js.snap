// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignSearch/> should equals to snapshot of SignSearch 1`] = `
<span
  className="sc-bdVaJa ioXKbX"
>
  <a
    href="/search/Search20Query"
    onClick={[Function]}
  >
    <i
      className="fa fa-search fa-1x pr-3"
      data-test="fa"
    />
  </a>
</span>
`;
